daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,316 @@
+	// -A INPUT -i <interface> -p <protocol> -m <protocol> --dport <port> -m comment --comment nordvpn -j ACCEPT
+	// -A INPUT -i <interface> -p <protocol> -m <protocol> --sport <port> -m comment --comment nordvpn -j ACCEPT
+	// -A OUTPUT -o <interface> -p <protocol> -m <protocol> --sport <port> -m comment --comment nordvpn -j ACCEPT
+	// -A OUTPUT -o <interface> -p <protocol> -m <protocol> --dport <port> -m comment --comment nordvpn -j ACCEPT
+	inputDportRule := fmt.Sprintf("INPUT -i %s -p %s -m %s --dport %d:%d -m comment --comment nordvpn -j ACCEPT", iface, protocol, protocol, portRange.min, portRange.max)
+	inputSportRule := fmt.Sprintf("INPUT -i %s -p %s -m %s --sport %d:%d -m comment --comment nordvpn -j ACCEPT", iface, protocol, protocol, portRange.min, portRange.max)
+	outputDportRule := fmt.Sprintf("OUTPUT -o %s -p %s -m %s --dport %d:%d -m comment --comment nordvpn -j ACCEPT", iface, protocol, protocol, portRange.min, portRange.max)
+	outputSportRule := fmt.Sprintf("OUTPUT -o %s -p %s -m %s --sport %d:%d -m comment --comment nordvpn -j ACCEPT", iface, protocol, protocol, portRange.min, portRange.max)
+
+	if f.enabled {
+		if err := f.commandExecutor.ExecuteCommand("-I " + inputDportRule); err != nil {
+			return fmt.Errorf("allowlisting input dport: %w", err)
+		}
+
+		if err := f.commandExecutor.ExecuteCommand("-I " + inputSportRule); err != nil {
+			return fmt.Errorf("allowlisting input sport: %w", err)
+		}
+
+		if err := f.commandExecutor.ExecuteCommand("-I " + outputDportRule); err != nil {
+			return fmt.Errorf("allowlisting output dport: %w", err)
+		}
+
+		if err := f.commandExecutor.ExecuteCommand("-I " + outputSportRule); err != nil {
+			return fmt.Errorf("allowlisting input dport: %w", err)
+		}
+	}
+
+	f.allowlistRules = append(f.allowlistRules, inputDportRule)
+	f.allowlistRules = append(f.allowlistRules, inputSportRule)
+	f.allowlistRules = append(f.allowlistRules, outputDportRule)
+	f.allowlistRules = append(f.allowlistRules, outputSportRule)
+
+	return nil
+}
+
+func (f *FirewallManager) SetAllowlist(udpPorts []int, tcpPorts []int, subnets []netip.Prefix) error {
+	ifaces, err := f.devices()
+	if err != nil {
+		return fmt.Errorf("listing interfaces: %w", err)
+	}
+
+	for _, subnet := range subnets {
+		for _, iface := range ifaces {
+			inputRule := fmt.Sprintf("INPUT -s %s -i %s -m comment --comment nordvpn -j ACCEPT", subnet.String(), iface.Name)
+			outputRule := fmt.Sprintf("OUTPUT -d %s -o %s -m comment --comment nordvpn -j ACCEPT", subnet.String(), iface.Name)
+
+			if f.enabled {
+				if err := f.commandExecutor.ExecuteCommand("-I " + inputRule); err != nil {
+					return fmt.Errorf("adding input accept rule for subnet: %w", err)
+				}
+				if err := f.commandExecutor.ExecuteCommand("-I " + outputRule); err != nil {
+					return fmt.Errorf("adding output accept rule for subnet: %w", err)
+				}
+			}
+
+			f.allowlistRules = append(f.allowlistRules, inputRule)
```

Comment:
same here.
Since there are more places, it might be easier to have some helper function that inserts a rule into the FW when enabled and appends into a list on success?

Reply:
Fixed, but not detected by GH as outdated.